lib/src/main/java/graphql/nadel/engine/blueprint/directives/NadelHydrationDirective.kt

```diff
@@ -0,0 +1,109 @@
+package graphql.nadel.engine.blueprint.directives
+
+import graphql.language.ArrayValue
+import graphql.language.ObjectField
+import graphql.language.ObjectValue
+import graphql.language.StringValue
+import graphql.nadel.util.AnyAstValue
+import graphql.schema.GraphQLAppliedDirective
+import graphql.schema.GraphQLFieldDefinition
+
+internal fun GraphQLFieldDefinition.isHydration(): Boolean {
+    return hasAppliedDirective(NadelHydrationDirective.SyntaxConstant.hydrated)
+}
+
+internal fun GraphQLFieldDefinition.getHydratedOrNull(): NadelHydrationDirective? {
```

Comment:
should we have another helper function that returns a list? For batch hydrations